src/physics.cpp

```diff
@@ -1052,50 +1029,37 @@ void sample_fission_neutron(int i_nuclide, const Reaction& rx, double E_in, Part
     // set the delayed group for the particle born from fission
     site->delayed_group = group;
 
-    int n_sample = 0;
-    while (true) {
-      // sample from energy/angle distribution -- note that mu has already been
-      // sampled above and doesn't need to be resampled
-      rx.products_[group].sample(E_in, site->E, mu, seed);
-
-      // resample if energy is greater than maximum neutron energy
-      constexpr int neutron = static_cast<int>(Particle::Type::neutron);
-      if (site->E < data::energy_max[neutron]) break;
-
-      // check for large number of resamples
-      ++n_sample;
-      if (n_sample == MAX_SAMPLE) {
-        // particle_write_restart(p)
-        fatal_error("Resampled energy distribution maximum number of times "
-          "for nuclide " + nuc->name_);
-      }
-    }
-
   } else {
     // ====================================================================
     // PROMPT NEUTRON SAMPLED
 
     // set the delayed group for the particle born from fission to 0
     site->delayed_group = 0;
+  }
 
-    // sample from prompt neutron energy distribution
-    int n_sample = 0;
-    while (true) {
-      rx.products_[0].sample(E_in, site->E, mu, seed);
-
-      // resample if energy is greater than maximum neutron energy
-      constexpr int neutron = static_cast<int>(Particle::Type::neutron);
-      if (site->E < data::energy_max[neutron]) break;
-
-      // check for large number of resamples
-      ++n_sample;
-      if (n_sample == MAX_SAMPLE) {
-        // particle_write_restart(p)
-        fatal_error("Resampled energy distribution maximum number of times "
-          "for nuclide " + nuc->name_);
-      }
+  // sample from prompt neutron energy distribution
+  int n_sample = 0;
+  double mu;
+  while (true) {
+    rx.products_[site->delayed_group].sample(E_in, site->E, mu, seed);
+
+    // resample if energy is greater than maximum neutron energy
+    constexpr int neutron = static_cast<int>(Particle::Type::neutron);
+    if (site->E < data::energy_max[neutron]) break;
+
+    // check for large number of resamples
+    ++n_sample;
+    if (n_sample == MAX_SAMPLE) {
+      // particle_write_restart(p)
```

Comment:
Did you intend to leave this commented out? It seems like it shouldn't be.

Reply:
This function doesn't actually have a reference to `p`, hence not being able to call it. If you'd like, I could remove this commented out line altogether